### PR TITLE
main, core: Core instantiate babe

### LINKED ISSUE #314: Reconcile types.BlockHeader and common.BlockHeader
right now we have two block headers, `types.BlockHeader` in core/types and `common.BlockHeader`. the one in `types` is used by the blocktree, the one in `common` is used by p2p/message.go.  The one in types has a hash field and the one in common doesn't.  It seems like we need both types, since a p2p BlockAnnounceMessage won't include a Hash field, and it makes sense for the block tree to know of the block hash for storage.

I suggest we make two types in `core/types`, `BlockHeader` and `BlockHeaderWithHash` so that we have both, and remove `common.BlockHeader`. Also `BlockHeader` should have a `Hash()` function.  Either that or we need to do custom encoding/decoding for BlockAnnounceMessage.

What are everyone's thoughts?

### LINKED ISSUE #364: Instantiating Babe Sessions
Presently the babe session is a parameter for the Core service, this should be updated such that the sessions are created inside of Core.

### PR DESCRIPTION
**Pull request by dziabko**
## Changes
- Moved the babe instantiation into the core service

## Tests:
```

```

### Issues:
closes #364 
closed #314

### CODE REVIEW

File: core/service.go [hunk 1]
```diff
@@ -34,21 +34,20 @@ var _ services.Service = &Service{}
 // It deals with the validation of transactions and blocks by calling their respective validation functions
 // in the runtime.
 type Service struct {
-	rt *runtime.Runtime
+	Rt *runtime.Runtime
```
Comment (noot) [inline core/service.go]: why is this exposed? I don't see it being used by any other packages

File: core/service.go [hunk 2]
```diff
@@ -34,21 +34,20 @@ var _ services.Service = &Service{}
...
 	b  *babe.Session
 
 	recChan  <-chan []byte
 	sendChan chan<- []byte
 }
 
 // NewService returns a Service that connects the runtime, BABE, and the p2p messages.
-func NewService(rt *runtime.Runtime, b *babe.Session, recChan <-chan []byte, sendChan chan<- []byte) *Service {
+func NewService(rt *runtime.Runtime, msgChan <-chan []byte) (*Service, error) {
```
Comment (noot) [inline core/service.go]: why is the sendChan being removed?

**Comment by ansermino** [comment]
This only removes the session from being passed in, *it does not instantiate a babe session*.

File: consensus/babe/babe.go [hunk 3]
```diff
@@ -107,7 +107,7 @@ func (b *Session) invokeBlockAuthoring() {
 			}
 
 			// Broadcast the block
-			blockAnnounceMsg := p2p.BlockAnnounceMessage{
+			blockAnnounceMsg := &p2p.BlockAnnounceMessage{
```
Comment (ansermino) [inline consensus/babe/babe.go]: I believe this needs to be the full header (BlockAnnounceMessage == Block Header)

**Pull request merged**